main_test.go

```diff
@@ -178,3 +174,20 @@ func TestParseTopicNamespace(t *testing.T) {
 		})
 	}
 }
+
+type mockProducer struct {
+	messages []*kafka.Message
+}
+
+func (p *mockProducer) Close() {
+}
+func (p *mockProducer) Flush(timeout int) int {
+	return 0
+}
+func (p *mockProducer) Produce(msg *kafka.Message, deliveryChan chan kafka.Event) error {
+	p.messages = append(p.messages, msg)
+	go func() {
+		deliveryChan <- msg
+	}()
+	return nil
+}
```

Comment:
I like this simple wrapper for testing purposes a lot!

Reply:
At first a omitted the goroutine and the entire testsuite locked up. 😁 Lucky I already expected I needed that. 😉 

Also, I ran the tests in `-race` mode, and we have some race-condition issues. 😅

Reply:
In the tests or in the implementation?

pg2kafka needs to run as a single process/thread (we need to read from the `outbound_event_queue` table in order), so it's likely not an issue.

Reply:
Cool! :+1: